File: DriveMateAndroid/app/src/main/java/com/drivemate/model/Message.kt
package com.drivemate.model

import java.util.Date
import java.util.UUID

enum class MessageRole {
    USER, ASSISTANT
}

data class PlaceAction(
    val id: String,
    val placeName: String,
    val address: String? = null,
    val rating: Double? = null,
    val placeId: String,
    val phoneNumber: String? = null
)

data class Message(
    val id: String = UUID.randomUUID().toString(),
    val role: MessageRole,
    val content: String,
    val timestamp: Date = Date(),
    val actions: List<PlaceAction> = emptyList()
)

File: DriveMate/DriveMate/Models/Message.swift
import Foundation

struct PlaceAction: Identifiable, Equatable {
    let id: String
    let placeName: String
    let address: String?
    let rating: Double?
    let placeId: String
    let phoneNumber: String?
}

struct Message: Identifiable, Equatable {
    let id: UUID
    let role: Role
    let content: String
    let timestamp: Date
    var actions: [PlaceAction]

    enum Role: String, Codable {
        case user
        case assistant
    }

    init(id: UUID = UUID(), role: Role, content: String, timestamp: Date = Date(), actions: [PlaceAction] = []) {
        self.id = id
        self.role = role
        self.content = content
        self.timestamp = timestamp
        self.actions = actions
    }
}
